### DISCUSSION TITLE
How to pass data to the frontend (Missing docs?)

### DISCUSSION THREAD

**Discussion by clorl** [oakserver/oak#616] [Q&A] [answered]
I'm looking for a "proper" way of sending backend data to frontend when serving the view. I know many templating engines do this, but is there a way to do it frontend-agnosticly?

[Context.state](https://deno.land/x/oak@v12.6.0/context.ts?s=Context#prop_state) seems to be for this exact purpose if I understand the docs correctly, but the doc doesn't go into how to retrieve this data on the frontend.

**Comment by kitsonk** [comment, marked as answer]
Oak is an http server framework, completely unopinionated and agnostic to what goes into a response.

A common pattern for sending data over HTTP would be to create [RESTful services](https://en.wikipedia.org/wiki/Representational_state_transfer) on the server, but this is only one approach to sending data to the client, of which oak has no direct opinion about how that is accomplished.
